### ISSUE TITLE
Publish Windows Server 2022 Envoy images 

### ISSUE THREAD

**Issue by davinci26** [envoyproxy/envoy#17656]
Windows Server 2022 comes with many new features but the most notable one is support for `SO_ORIGINAL_DST` and traffic redirection via HNS policies. These features are an absolute requirement to be able to run Envoy as a sidecar in a service mesh solution.

We should publish windows server 2022 images of Envoy (**ideally** based on Nanoserver)

Limitations:

1. windows-server-latest CI is not compatible with window server 2022 which means that we can't build the image from a windows host
2. `docker buildx` is not compatible with `RUN` commands which we use to set the path.

Problem (1) is not possible to solve, so we will have to solve (2).

We use `RUN` commands for two reasons:

1. Setting up the Envoy user, which can be moved into a runtime script.
2. Place `envoy.exe` into a location that is already in the path and thus remove the need for running `setx` commands.

tl;dr: We should build Windows images from linux with `docker buildx`

**Comment by davinci26** [comment]
@wrowe and @phlax this come up on the windows-dev channel. I think that this is going to be a big win on many different levels

**Comment by davinci26** [comment]
@thhous-msft this should be a good issue if you find it interesting

cc @envoyproxy/windows-dev

**Comment by sunjayBhatia** [comment]
If you need CI/build tools for Windows Server 2022 might want to also make a tracking issue for envoy-build-tools, might be good to get started for the future but probably not a blocker to just publish builds

**Comment by davinci26** [comment]
I want to solicit some feedback here on the implementation that I am thinking of here:

### Problem 

The biggest problem to port the Windows Dockerfile to use buildx is the usage of 
```
USER ContainerAdministrator
RUN net user /add "EnvoyUser"
RUN net localgroup "Network Configuration Operators" "EnvoyUser" /add
```

This part is really important though for all cases Envoy runs as a sidecar and it **needs** to be running on as `EnvoyUser`

### Proposed Solution

tl;dr: Move the user creation to the container startup.

That would work in the following way:

1. Create a new executable `Envoy-privilege-dropper` which does the following
1.1 Create `EnvoyUser`
1.2 Add `EnvoyUser` to `Network Configuration Operators`
1.3 Create a **child process** to run Envoy as `EnvoyUser`
1.4 Redirect the standard output/error of child process to the parent process

I am positive that the above solution should work but I think that this additional layer of abstraction might complicate things in terms of debugging.
